### PR TITLE
fix(core/idl): linking attribute id/type with same name

### LINKED ISSUE #1768: Multiple dfn with the same name
## Important info

* URL to affected spec: https://w3c.github.io/hr-time
* ReSpec version: 22.3.1

## Description of problem
The spec has a warning:

`Missing <dfn> for Performance interface.`

In the spec, we have:

- A  `<dfn>Performance</dfn>` in a section with `data-dfn-for="Performance"  data-link-for="Performance"`.

- A `<dfn>performance</dfn>` in a section with `data-dfn-for="WindowOrWorkerGlobalScope"  data-link-for="WindowOrWorkerGlobalScope"`.

It seems that the two dfns being the same name is causing the problem. In particular, there are two problems:

1.  "Performance" title in section 5 is not orange (but fortunately links to it do work).
2. The link to "Performance" in section 6 in `readonly    attribute Performance performance;` is to the attribute instead of the interface (even if I try to add a data-link-for for "Performance").

Not sure if this is an issue with ReSpec or if there's a way to fix this?

### PR DESCRIPTION
**Pull request by marcoscaceres**
closes #1768

This prevents the attribute's type from using the parent element's `data-link-for`, which was confusing the linker. Types always need to be defined at top level things (i.e., they are not "for" things), so this should work.

### CODE REVIEW

**Comment by marcoscaceres** [comment]
Fixes methods too :) Adding a test.

**Comment by saschanaz** [comment]
Does this also fix #1508?

File: tests/spec/core/webidl-spec.js [hunk 1]
```diff
@@ -321,10 +321,12 @@ describe("Core - WebIDL", function() {
   attribute FrozenArray<DOMString> alist;
   // 4.0
   attribute Promise<DOMString> operation;
-};`;
+  // 5.0
+  readonly attribute Performance performance;
+};`.trim();
```
Comment (saschanaz) [inline tests/spec/core/webidl-spec.js]: nit: this `trim()` doesn't seem to be needed

**Comment by marcoscaceres** [comment]
> Does this also fix #1508?

I'm not sure, tbh. I'm going to say, "probably", but will need to do some more investigation/testing.

File: tests/spec/core/webidl-spec.js [hunk 2]
```diff
@@ -352,6 +354,22 @@ describe("Core - WebIDL", function() {
         .find(c => c.textContent.includes("alist"))
         .querySelector(".idlAttrName a")
     ).toBeNull();
+
+    const performanceIterfaceLink = Array.from(
```
Comment (saschanaz) [inline tests/spec/core/webidl-spec.js]: nit: `iterface` 😆

Comment (marcoscaceres) [reply tests/spec/core/webidl-spec.js]: heh, almost was to keep it :)

**Comment by marcoscaceres** [comment]
> Well I was wrong with "case sensitive definitionMap" as solution. Always a simpler way!

Definitely not wrong. I think we should, in the future, treat IDL things as case sensitive. IDL is case sensitive, so it makes sense to do so.

**Comment by marcoscaceres** [comment]
@sidvishnoi, the rebase look good! Thanks for doing it!

**Pull request merged**